### ISSUE TITLE
FindNetcdf

### ISSUE THREAD

**Issue by mmiesch** [JCSDA/ecbuild#23]
Modify FindNetcdf module to be able to find static netcdf libraries

**Comment by aerorahul** [comment]
@mmiesch My solution to this is to use `FindNetCDF4.cmake` logic in `FindNetCDF.cmake`.  This takes us further from ecmwf::develop. 
@markjolah wants to use the newer `cmake` support, and I agree, but I am not sure when/who will do that.  @markjolah had volunteered in the past.  I can assist, if needed, but he is the expert.
So either we do what @markjolah suggests, or we discuss with ECMWF and see if they are open to the alternative.
Thoughts?

**Comment by markjolah** [comment]
I'll take another look at this.  I was waiting until we had some direction on branches for this repo.   Are we back to using `JCSDA/develop` branch?

**Comment by mmiesch** [comment]
Yes - we're back to using the ```JCSDA/develop``` branch which is, right now, the same as the ```3.1.0.jcsda3``` release

**Comment by markjolah** [comment]
OK.  I looked over the new FindNetCDF.cmake and it looks much improved, and we definitely want to use this new version.  Now, we need to figure out what is currently not working.  
It looks to me like the CMake code will prefer to find shared libraries over static, but it should be capable of finding static libraries too.  Is the problem that shared libraries are found, but you really wanted static libraries, or that no libraries are found at all in environments where only static libraries are available?  These scenarios will have different possible solutions.  Also if there is a particular machine/environment it is not working on that I can debug it there?

**Comment by aerorahul** [comment]
@markjolah 
If you can get the new FindNetCDF.cmake to work on Discover using the netCDF from Baselibs, I'ld say it will be fool-proof and robust.

The issue is the netCDF dependencies such as HDF5, HDF4, Curl, etc. nc-config finds all of those with the appropriate call.

**Comment by markjolah** [comment]
Hmm..  I'm looking for baselibs modules and I am only seeing them under your modules dir at:
```
/discover/nobackup/projects/gmao/obsdev/rmahajan/opt/modulefiles
```
Are these the modules we want to work?  Is there a master module to load to get started?

**Comment by aerorahul** [comment]
@markjolah 
Baselibs does not have a module unfortunately.
I created this modulefule to make life easy for us to use Baselibs, without having to install everything from scratch and making sure the model (GEOS) and JEDI use the same software stack.

Basically, baselibs installs under a flat structure
bin, lib, include, share, etc ...

**Comment by markjolah** [comment]
OK.  I had a look around at the baselibs.  It looks like the latest version with support for intel-17.0.7.259 is `ESMA-Baselibs-5.1.4` ? All the others are using intel-18 which I know is problematic for JEDI.  I'm seeing the libraries at:
```
 /discover/swdev/mathomp4/Baselibs/ESMA-Baselibs-5.1.4/x86_64-unknown-linux-gnu/ifort_17.0.7.259-intelmpi_17.0.7.259/Linux/lib
```
I also looked through your module files.  Is it possible that you just need to update your cmake??  You are force loading `other/cmake-3.8.2` which is ancient.  Discover has `other/cmake-3.15.5` available.  That will solve any FindMPI problems you were having, and it will also allow you to more effectively use environment variables to control how find_library() looks for packages.

Notice the documentation changes betewen:

**Comment by aerorahul** [comment]
@markjolah 
We can use the newer cmake; sure.

The Intel-17 build of Baselibs is being used for JEDI development.  The GEOS model uses a later version (Intel-18).  Both will hopefully be reconciled in Intel-19 or later.

**Comment by mmiesch** [comment]
It's possible jedi may work with intel 18 now.  We had to change the code to get it to work with intel 19.  I don't know if anyone has tried intel 18 again after those changes.   Those fixes might have fixed the problems with intel18 too.

**Comment by markjolah** [comment]
Addressed in #24
